lib/stages/FEngine.cpp

```diff
@@ -0,0 +1,531 @@
+#include <Config.hpp>
+#include <Stage.hpp>
+#include <StageFactory.hpp>
+#include <cassert>
+#include <chimeMetadata.hpp>
+#include <chordMetadata.hpp>
+#include <complex>
+#include <cstdint>
+#include <cstring>
+#include <fstream>
+#include <julia.h>
+#include <juliaManager.hpp>
+#include <string>
+#include <vector>
+#include <visUtil.hpp>
+
+#if !KOTEKAN_FLOAT16
+#warning "The F-Engine simulator requires float16 support"
+#else
+
+class FEngine : public kotekan::Stage {
```

Comment:
We should have some documentation on what this does and generically how it works (including what config parameters it takes).   Also generally we've used separate header files for declarations, could we do that for consistency?